charts/partials/templates/_metadata.tpl

```diff
@@ -1,5 +1,5 @@
 {{- define "partials.annotations.created-by" -}}
-linkerd.io/created-by: {{ .Values.cliVersion | default (printf "linkerd/helm %s" (.Values.cniPluginVersion | default .Values.linkerdVersion)) }}
+linkerd.io/created-by: {{ .Values.cliVersion | default (printf "linkerd/helm %s" (.Values.image.version | default .Values.linkerdVersion)) }}
```

Comment:
Tests referencing this fails with a nil error, how can I scope this to the cni Values?

Reply:
I think you can check for existence using the `ternary` function, something like this:
```
linkerd.io/created-by: {{ .Values.cliVersion | default (printf "linkerd/helm %s" (.Values.image | ternary  .Values.image.version .Values.linkerdVersion)) }}
```

Reply:
Hmm, it seems that the value exists but not in this `.Values` object and I'm not sure how to reference the cni Values object from here.

Reply:
in other words, the nil pointer is just a symptom of the scoping issue and not the exact thing I need to fix.

Reply:
I don't see why that `.Values` wouldn't be in scope there... although this `ternary` is throwing some errors for me. The following tests fine for me, leveraging Alex' trick from #10235:
```
linkerd.io/created-by: {{ .Values.cliVersion | default (printf "linkerd/helm %s" ((.Values.image).version | default .Values.linkerdVersion )) }}
```

Reply:
Thanks! That seems to be working.